src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -147,7 +147,11 @@ export class DefaultTelemetryService implements TelemetryService {
             // this is async so that we don't have pseudo-concurrent invocations of the callback
             async () => {
```

Comment:
@rli regarding your comment on line 147: I looked at the git logs and didn't find any explication, do you remember the motivation? 

- Why do we want to avoid concurrent invocations of the callback? 
- And how does `async` avoid that?

Reply:
I think the idea is that `flushRecords` may take longer than the timeout period, so we shouldn't refresh the timer until that completes. Since that's an `await` call, the timer callback needs to be `async`.